slackclient/client.py

```diff
@@ -86,20 +172,40 @@ def api_call(self, method, timeout=None, **kwargs):
                 u'{"ok":false,"error":"channel_not_found"}'
 
             See here for more information on responses: https://api.slack.com/web
-        '''
-        response_body = self.server.api_call(method, timeout=timeout, **kwargs)
+        """
+        # Check for missing or expired access token before submitting the request
+        if method != 'oauth.access' and self.refresh_token:
+            current_ts = int(time.time())
+            expired_token = current_ts > self.access_token_expires_at
+            if (expired_token or self.token is None):
+                self.refresh_access_token()
+
+        response_body = self.server.api_call(self.token, request=method, timeout=timeout, **kwargs)
+
+        # Attempt to parse the response as JSON
         try:
             result = json.loads(response_body)
         except ValueError as json_decode_error:
             raise ParseResponseError(response_body, json_decode_error)
+        response_json = json.loads(response_body)
 
-        if "ok" in result and result["ok"]:
+        if result["ok"]:
```

Comment:
check for existence of `"ok"` before looking at the value (otherwise this code might raise a `KeyError`).

Reply:
it seems like any time we do a dict key lookup, we should void direct indexing with keys and instead use the `.get(key, default_value)` method that all dicts have. in fact, if we could enforce this with a linter rule, that would be a big win!